crates/core/tedge_agent/src/file_transfer_server/actor.rs

```diff
@@ -141,31 +144,26 @@ mod tests {
     }
 
     #[tokio::test]
-    #[serial_test::serial]
-    async fn check_server_does_not_panic_when_port_is_in_use() -> Result<(), anyhow::Error> {
+    async fn check_server_does_not_panic_when_port_is_in_use() -> anyhow::Result<()> {
         let ttd = TempTedgeDir::new();
 
         let http_config = HttpConfig::default()
             .with_data_dir(ttd.utf8_path_buf().into())
             .with_port(3746);
```

Comment:
suggestion: This will fail if port 3746 is already bound

I know this is previous behaviour, but seeing as how you improved the rest of the test, could you also change this to bind on port 0 and then query which port was actually bound to use later? If it's annoying to query the port after the port was bound, then that's okay, I don't think we use this port anywhere else, so it won't be a problem when running tests in parallel.

Reply:
It's a good point, and actually means all the complexity can go away, as the test could be simplified to "create a tcp listener" then "spawn a server with a timeout" and check we get an error, which removes the race issue entirely (and the assumptions that make `biased` correct).